### PR TITLE
Feat 123 update nsb2023

### LINKED ISSUE #123: Handle sharepoint 2023 db schema changes
From Rob: 

>  For the 2023 database the fields below are now in number format. We ended up breaking out the D/V into 3 fields to account for multiple depths. Finally we now have the option for 6 burr holes so # can be 1-6.
>> Burr # Angle
>> Burr # A/P
>>Inj#volperdepth
>>Burr # M/L
>>Burr # D/V 1
>>Burr # D/V 2
>>Burr # D/V 3
>>Fiber Implant D/V

### PR DESCRIPTION
**Pull request by mekhlakapoor**
closes #123 

This PR handles the changes in the NSB 2023 database. Some of these changes include numerical fields (such as ML, AP, DV) being floats instead of strings. They also added about 75 fields (burr 5 and 6 and all respective data fields, each burr has 3 dv fields now that get mapped to a list, behavior fields, etc). 

This PR: 
- regenerate models and mapping for nsb2023
- remove str parsers for number format fields
- maps a list of depth coordinates for dv (fore each burr hole)
- maps burr holes 5 and 6
- handle other changes (like headframe mapping)
- update to latest aind-data-schema version

### CODE REVIEW

**Review by jtyoung84 — CHANGES_REQUESTED** [review]
It will be easier to remove if you remove all the unnecessary changes that are being made. For example, the test files look like they're just adding an extra space. Also, make sure the aind-data-schema is bumped up to the current version number.

File: src/aind_metadata_service/sharepoint/nsb2023/mapping.py [hunk 1]
```diff
@@ -286,32 +245,52 @@ def _parse_virus_strain_str(_: Optional[str]) -> Optional[str]:
...
-            if parsed_string is not None:
-                return self._parse_basic_float_str(parsed_string.group(1))
-            else:
-                return None
-        else:
-            return None
-
     @property
-    def aind_age_at_injection(self) -> Optional[float]:
+    def aind_age_at_injection(self) -> Optional[str]:
```
Comment (mekhlakapoor) [inline src/aind_metadata_service/sharepoint/nsb2023/mapping.py:249]: float not str

File: src/aind_metadata_service/sharepoint/nsb2023/mapping.py [hunk 2]
```diff
@@ -286,32 +245,52 @@ def _parse_virus_strain_str(_: Optional[str]) -> Optional[str]:
...
+            }.get(self._nsb.behavior, None)
+        )
+
+    @property
+    def aind_behavior_complete(self) -> Optional[datetime]:
+        """Maps behavior_complete to aind model"""
+        return self._nsb.behavior_complete
+
+    @property
+    def aind_behavior_type(self) -> Optional[Any]:
```
Comment (mekhlakapoor) [inline src/aind_metadata_service/sharepoint/nsb2023/mapping.py:281]: return None

File: src/aind_metadata_service/sharepoint/nsb2023/mapping.py [hunk 3]
```diff
@@ -411,7 +390,7 @@ def aind_burr2_injection_devi(self) -> Optional[Any]:
         )
 
     @property
-    def aind_burr2_perform_during(self) -> Optional[During]:
+    def aind_burr2_perform_during(self) -> Optional[Any]:
```
Comment (mekhlakapoor) [inline src/aind_metadata_service/sharepoint/nsb2023/mapping.py:393]: output type should be During

File: src/aind_metadata_service/sharepoint/nsb2023/mapping.py [hunk 4]
```diff
@@ -1262,23 +1687,25 @@ def aind_id(self) -> Optional[int]:
...
-                self._nsb.implant_id_coverslip_type.N_3002: None,
-                self._nsb.implant_id_coverslip_type.N_3MM: None,
-                self._nsb.implant_id_coverslip_type.N_3MM_STACKED_COVERSLIP: None,
-                self._nsb.implant_id_coverslip_type.N_5MM_STACKED_COVERSLIP: None,
-                self._nsb.implant_id_coverslip_type.N_5MM_STACKED_COVERSLIP_W: (
-                    None
-                ),
-            }.get(self._nsb.implant_id_coverslip_type, None)
-        )
+        return {
```
Comment (jtyoung84) [inline src/aind_metadata_service/sharepoint/nsb2023/mapping.py:1690]: You'll need to add this again:
```
return (
            None
            if self._nsb.implant_id_coverslip_type is None
            else {
```

File: src/aind_metadata_service/sharepoint/nsb2023/mapping.py [hunk 5]
```diff
@@ -1479,12 +1906,114 @@ def aind_inj4ret_setting(self) -> Optional[Any]:
...
+
+    @property
+    def aind_inj6_storage_location(self) -> Optional[str]:
+        """Maps inj6_storage_location to aind model"""
+        return self._nsb.inj6_storage_location
+
+    @property
+    def aind_inj6_type(self) -> Optional[InjectionType]:
+        """Maps inj6_type to aind model"""
+        return {
```
Comment (jtyoung84) [inline src/aind_metadata_service/sharepoint/nsb2023/mapping.py:1989]: You'll need the `return None if` here

File: src/aind_metadata_service/sharepoint/nsb2023/mapping.py [hunk 6]
```diff
@@ -1479,12 +1906,114 @@ def aind_inj4ret_setting(self) -> Optional[Any]:
...
+
+    @property
+    def aind_inj6_virus_strain_rt(self) -> Optional[str]:
+        """Maps inj6_virus_strain_rt to aind model"""
+        return self._nsb.inj6_virus_strain_rt
+
+    @property
+    def aind_inj6ret_setting(self) -> Optional[Any]:
+        """Maps inj6ret_setting to aind model"""
+        return {
```
Comment (jtyoung84) [inline src/aind_metadata_service/sharepoint/nsb2023/mapping.py:2003]: You'll need the return None if here

**Review by jtyoung84 — CHANGES_REQUESTED** [review]

File: src/aind_metadata_service/sharepoint/nsb2023/mapping.py (hunk 6 above)
Comment (mekhlakapoor) [reply src/aind_metadata_service/sharepoint/nsb2023/mapping.py:2003]: agh, I thought I caught them all this time. Thanks for finding these

Comment (mekhlakapoor) [reply src/aind_metadata_service/sharepoint/nsb2023/mapping.py]: i went through the mapping and (hopefully) caught everything this time.

**Pull request merged**